src/tir/transforms/lower_init_block.cc

```diff
@@ -0,0 +1,85 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*!
+ * Lower block init stmt into branch stmt
+ * \file lower_reduction.cc
+ */
+#include <tvm/tir/op.h>
+#include <tvm/tir/stmt_functor.h>
+#include <tvm/tir/transform.h>
+
+namespace tvm {
+namespace tir {
+
+class InitBlockLower : public StmtMutator {
+ private:
+  Stmt VisitStmt_(const BlockNode* block) final {
+    if (!block->init.defined()) {
+      return StmtMutator::VisitStmt_(block);
+    }
+    Stmt init = RealizeInitBlock(block->init.value(), block->iter_vars);
+    Stmt body = VisitStmt(block->body);
+    auto n = CopyOnWrite(block);
+    n->init = NullOpt;
+    n->body = SeqStmt::Flatten(init, body);
+    return Block(n);
+  }
+
+  static Stmt RealizeInitBlock(const Stmt& init, const Array<IterVar>& iter_vars) {
```

Comment:
Could you clarify the meaning of the word "realize" here?

Reply:
What about "concretize"? My understanding is that the init-block is something "conceptual" for initializing "zero point" of a reduction, and this pass is used to make it concrete, lowering to "if-then-else" statement

Reply:
I got your point, and I feel "LowerInitBlock" itself is the best name to describe what it does, so maybe the simplest way is manually inlining this function so that we won't be bothered by its name, or just give it a general helper name to let people know this is just the implementation of lowering init block.

Reply:
This helper is something a bit standalone (making the init block to If-Then-Else), and inlining this function might make the logic in the function above a bit vague, so what about we just call it "DoLowering"?

Reply:
DoLowering sounds good to me.